packages/nimble-components/src/table/components/cell/index.ts

```diff
@@ -14,10 +20,49 @@ declare global {
  * A styled cell that is used within the nimble-table-row.
  * @internal
  */
-export class TableCell extends FoundationElement {
-    // TODO: This should be replaced with an instance of TableCellState<TCellData>
+export class TableCell<
+    TCellData extends TableRecord = TableRecord
+> extends FoundationElement {
     @observable
-    public data: TableFieldValue;
+    public data?: TableCellState<TCellData>;
+
+    @observable
+    public cellTemplate?: ViewTemplate;
+
+    @observable
+    public cellStyles?: ElementStyles;
+
+    /**
+     * @internal
+     */
+    public readonly cellContainer: HTMLElement | undefined = undefined;
+
+    private customCellView?: HTMLView = undefined;
+
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.customCellView = this.cellTemplate?.render(
+            this.data,
+            this.cellContainer!
+        );
```

Comment:
Should this work be done in a cellTemplateChanged method instead / in addition?

At least only within the context of this element it's making the assumption that cellTemplate is always set before connected, which in an isolated context doesn't seem right. The cellTemplate could update after conencted.

Seems like updating the cellTemplate should also bind the data while updating data just binds the data.

Reply:
I've now added the same cell to render the template in `cellTemplateChanged`.